homeassistant/components/vicare/fan.py

```diff
@@ -0,0 +1,172 @@
+
+        with suppress(PyViCareNotSupportedFeatureError):
+            self._attributes["vicare_modes"] = self._api.getAvailableModes()
+        self._attributes["vicare_programs"] = self._api.getAvailablePrograms()
+
+    def update(self) -> None:
+        """Update state of fan."""
+        try:
+            with suppress(PyViCareNotSupportedFeatureError):
+                self._attributes["active_vicare_mode"] = self._api.getActiveMode()
+            with suppress(PyViCareNotSupportedFeatureError):
+                self._attributes["active_vicare_program"] = self._api.getActiveProgram()
+        except RequestConnectionError:
+            _LOGGER.error("Unable to retrieve data from ViCare server")
+        except ValueError:
+            _LOGGER.error("Unable to decode data from ViCare server")
+        except PyViCareRateLimitError as limit_exception:
+            _LOGGER.error("Vicare API rate limit exceeded: %s", limit_exception)
+        except PyViCareInvalidDataError as invalid_data_exception:
+            _LOGGER.error("Invalid data from Vicare server: %s", invalid_data_exception)
+
+    def turn_off(self, **kwargs: Any) -> None:
+        """Turn the entity off."""
+        raise ServiceValidationError(
+            translation_domain=DOMAIN,
+            translation_key="fan_must_be_on",
+        )
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return true if the entity is on."""
+        # Viessmann ventilation unit cannot be turned off
+        return True
+
+    @property
+    def percentage(self) -> int | None:
+        """Return the current speed percentage."""
+        return ordered_list_item_to_percentage(
+            ORDERED_NAMED_FAN_SPEEDS, self._attributes["active_vicare_program"]
+        )
+
+    def set_percentage(self, percentage: int) -> None:
+        """Set the speed of the fan, as a percentage."""
+
+        if self._attributes["active_vicare_mode"] != str(VentilationMode.PERMANENT):
+            self.set_preset_mode(VentilationMode.PERMANENT)
+
+        level = percentage_to_ordered_list_item(ORDERED_NAMED_FAN_SPEEDS, percentage)
+        _LOGGER.debug("changing ventilation level to %s", level)
+        self._api.setPermanentLevel(level)
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode, e.g., auto, smart, interval, favorite."""
+
+        return self._attributes["active_vicare_mode"]
+
+    @property
+    def preset_modes(self) -> list[str] | None:
+        """Return a list of available preset modes."""
+        return list[str](PRESET_MODES)
+
+    def set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        _LOGGER.debug("changing ventilation mode to %s", preset_mode)
+        self._api.setActiveMode(preset_mode)
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Show Device Attributes."""
+        return self._attributes
```

Comment:
Generally we don’t add extra state attributes to new entities. They should instead be first class sensor entities with proper device classes and descriptions

Reply:
Unless, there is a good reason to keep them here (which generally only means they have no meaning on their own as a separate sensor).